src/transpiler.ts

```diff
@@ -111,9 +111,9 @@ function applyPasses(ast: AST, options: TranspilationOptions & PrintOptions): AS
     ['U', UnloadingAssignment],
     ['V', VariableDeclarationInitialiser],
     ['Vs', VariableDeclarationExpressionSplitter],
+    ['Rf', References],
     ['I', ImplicitConversionToExplicit],
     ['Dh', DeleteHandler],
-    ['Rf', References],
```

Comment:
DeleteHandler must go before References because the way we currently handle `memory` type deletion is by creating a new assignation to the default value, and then the Reference pass allocates it.

Reply:
We also need References after ImplicitToExplicit for array state variable initialisation to work properly. The rvalue array needs to have been converted to the right type before the References pass allocates the space for that type